graphql/schemabuilder/types.go

```diff
@@ -97,21 +97,75 @@ func BatchFilterFieldWithFallback(name string, batchFilter interface{}, filter i
 		Options:         options,
 	}
 	var fieldFuncTextFilterFields fieldFuncOptionFunc = func(m *method) {
-		if m.TextFilterFuncs == nil {
-			m.TextFilterFuncs = map[string]Filter{}
+		if m.TextFilterFuncsStruct == nil {
+			m.TextFilterFuncsStruct = map[string]Filter{}
 		}
-		if filter, ok := m.TextFilterFuncs[name]; ok {
-			panic("Batch Field Filters have the same name: " + filter.Name)
+		if filter, ok := m.TextFilterFuncsStruct[name]; ok {
+			panic("Filters Fields have the same name: " + filter.Name)
 		}
-		m.TextFilterFuncs[name] = filterStruct
+		m.TextFilterFuncsStruct[name] = filterStruct
 	}
 	return fieldFuncTextFilterFields
 }
 
-type SortFields map[string]interface{}
+type Sort struct {
+	Name          string
+	SortFunc      interface{}
+	BatchSortFunc interface{}
+	FallbackFlag  func(context.Context) bool
+}
 
-func (s SortFields) apply(m *method) {
-	m.SortFuncs = s
+func SortField(name string, sort interface{}) FieldFuncOption {
+	var sortStruct = Sort{
+		Name:     name,
+		SortFunc: sort,
+	}
+	var fieldFuncSortFields fieldFuncOptionFunc = func(m *method) {
+		if m.SortFuncsStruct == nil {
+			m.SortFuncsStruct = map[string]Sort{}
+		}
+		if sort, ok := m.SortFuncsStruct[name]; ok {
+			panic("Sorts fields have the same name: " + sort.Name)
+		}
+		m.SortFuncsStruct[name] = sortStruct
+	}
+	return fieldFuncSortFields
+}
+
+func BatchSortField(name string, batchSort interface{}) FieldFuncOption {
+	var sortStruct = Sort{
+		Name:          name,
+		BatchSortFunc: batchSort,
+	}
+	var fieldFuncSortFields fieldFuncOptionFunc = func(m *method) {
+		if m.SortFuncsStruct == nil {
+			m.SortFuncsStruct = map[string]Sort{}
+		}
+		if filter, ok := m.SortFuncsStruct[name]; ok {
+			panic("Batch sorts fields have the same name: " + filter.Name)
+		}
+		m.SortFuncsStruct[name] = sortStruct
+	}
+	return fieldFuncSortFields
+}
+
+func BatchSortFieldWithFallback(name string, batchSort interface{}, sort interface{}, flag func(context.Context) bool) FieldFuncOption {
+	var sortStruct = Sort{
```

Comment:
wait, should these just return `&method{` structs?  Is there any reason for us to hold onto a specific `Sort{` struct? (maybe follow-up)